### PR TITLE
run_tests: dedicate a container per test file for integration tests

### PR DESCRIPTION
**Pull request by aquilamacedo**
This PR refactors the integration test setup to dedicate a separate
container for each test file. The `prepare_testing_environment` function
is introduced in individual test scripts to set up the environment
specifically for each test, ensuring isolated test environments.

### CODE REVIEW

File: run_tests.sh [hunk 1]
```diff
@@ -125,12 +113,10 @@ function run_tests()
...
-  if [[ "${integration_tests_setup}" == 1 ]]; then
-    printf '\n' # add new line after the last "OK"
-    say 'Tearing down containers used in integration tests...'
-    teardown_containers
-    printf '\n'
-  fi
+  printf '\n' # add new line after the last "OK"
+  say 'Tearing down containers used in integration tests...'
+  teardown_containers
+  printf '\n'
```
Comment (rodrigosiqueira) [inline run_tests.sh:119]: Can we try moving the first `printf` into the `say` function? Also, consider moving the last `printf` to the `teardown_containders`  function.

File: tests/integration/device_test.sh [hunk 2]
```diff
@@ -4,6 +4,11 @@ include './tests/unit/utils.sh'
 include './tests/integration/utils.sh'
 include './src/device_info.sh'
 
+function oneTimeSetUp()
+{
+  prepare_testing_environment 'device'
```
Comment (rodrigosiqueira) [inline tests/integration/device_test.sh:9]: With this function, we display the setup for every test, for example:

```
Preparing environment for kw config integration tests...
[1/6] Using cached container image for archlinux.
[2/6] Creating archlinux container.
[3/6] Using cached container image for debian.
[4/6] Creating debian container.
[5/6] Using cached container image for fedora.
[6/6] Creating fedora container.
```

Can we add a silent option for the `setup_container_environment` function?

**Review by rodrigosiqueira — CHANGES_REQUESTED** [review]

File: tests/integration/device_test.sh (hunk 2 above)
Comment (aquilamacedo) [reply tests/integration/device_test.sh]: I've included a `--verbose` option for situations where we need to see this information.

**Comment by rodrigosiqueira** [comment]
Hi @aquilamacedo , thanks for your PR!
Change merged into the unstable branch.

**Pull request closed without merging**